Foundation/JSONSerialization.swift

```diff
@@ -738,11 +738,20 @@ private struct JSONReader {
             return nil
         }
 
-        if !UTF16.isLeadSurrogate(codeUnit) {
+        let isLeadSurrogate = UTF16.isLeadSurrogate(codeUnit)
+        let isTrailSurrogate = UTF16.isTrailSurrogate(codeUnit)
+
+        if !isLeadSurrogate && !isTrailSurrogate {
             return (String(UnicodeScalar(codeUnit)!), index)
         }
 
-        guard let (trailCodeUnit, finalIndex) = try consumeASCIISequence("\\u", input: index).flatMap(parseCodeUnit) , UTF16.isTrailSurrogate(trailCodeUnit) else {
+        guard !isTrailSurrogate else {
```

Comment:
Could you rewrite as `guard isLeadSurrogate else {`? The double negatives made me read this a couple times to verify it was actually correct.

Reply:
Fair enough. Also added some comments that I think make the code a bit clearer for a person that is not a Unicode expert (like me).